sample-app/src/main.js

```diff
@@ -83,12 +83,15 @@ const questions1 = [
   },
   {
       type: "input",
+      guiOptions: {
+        applyDefault: true
+      },
       name: "notes",
       message: function (answers) {
           return `Information about ${answers.name}`;
       },
       default: function (answers) {
-          return `Information about ${answers.name}`;
+          return answers.name === "Joe" ? `Information about ${answers.name}` : `Details about ${answers.name}`;
```

Comment:
What were you trying to do here?

You could have 2 answers with a default value: 1 with and 1 without `guiOptions.applyDefaultWhenDirty`